glom/core.py

```diff
@@ -1431,6 +1448,17 @@ def __call__(self, *args, **kwargs):
             # TODO: typecheck kwarg vals?
         return _t_child(self, '(', (args, kwargs))
 
+    def __star__(self):
+        return _t_child(self, 'x', None)
```

Comment:
sticking to the singleton idea above, args here would become `(self, 'P', STAR)` or similar.

Reply:
hrm............ that could work, it would be a break with the pattern of the code in `_t_teval` -- instead of

```python
elif op == 'x':
   recurse()
elif op == 'P':
   get(target, arg)
```

something more like

```python
elif op == 'P':
   if arg is STAR:
       recurse()
   else:
      get(arget, arg)
```

I can't point to any specific problem, it just seems simpler to keep "code" and "data" separate, rather than having special values in the `arg` position

In this case, I think as you mentioned in the same comment `Path(T.__star__())` would do the right thing without needing to change the operator `'x'` into a special control value for the operator `'P'`

Reply:
I've got this idea for how to further optimize `T` (this would be a separate PR) -- right now it is laid out like this:

```python
(T, op, arg, op, arg, ...)
```

I'm thinking about a different structure:

```python
(T, op-string, arg, arg, arg, ...)
```

Advantages:

* multi-byte string is FAR more compact than tuple-of-1-byte-strings
* operators can have different #'s of args freely (many have no argument and None arg; call() has 3 args which require a sub-tuple)